parsons/google/google_cloud_storage.py

```diff
@@ -547,29 +551,82 @@ def unzip_blob(
             String representation of decompressed GCS URI
         """
 
+        compression_params = {
+            "zip": {
+                "file_extension": ".zip",
+                "compression_function": self.__zip_decompress_and_write_to_gcs,
+                "read": "r",
+            },
+            "gzip": {
+                "file_extension": ".gz",
+                "compression_function": self.__gzip_decompress_and_write_to_gcs,
+            },
+        }
+
+        file_extension = compression_params[compression_type]["file_extension"]
+        compression_function = compression_params[compression_type][
+            "compression_function"
+        ]
+
         compressed_filepath = self.download_blob(
             bucket_name=bucket_name, blob_name=blob_name
         )
 
-        decompressed_filepath = compressed_filepath.replace(".gz", "")
+        decompressed_filepath = compressed_filepath.replace(file_extension, "")
         decompressed_blob_name = (
-            new_filename if new_filename else blob_name.replace(".gz", "")
+            new_filename if new_filename else blob_name.replace(file_extension, "")
         )
         if new_file_extension:
             decompressed_filepath += f".{new_file_extension}"
             decompressed_blob_name += f".{new_file_extension}"
 
         logger.debug("Decompressing file...")
+        compression_function(
+            compressed_filepath=compressed_filepath,
+            decompressed_filepath=decompressed_filepath,
+            decompressed_blob_name=decompressed_blob_name,
+            bucket_name=bucket_name,
+            new_file_extension=new_file_extension,
+        )
+
+        return self.format_uri(bucket=bucket_name, name=decompressed_blob_name)
+
+    def __gzip_decompress_and_write_to_gcs(self, **kwargs):
```

Comment:
Dig the two different, private functions for the two different cases here

Wonder if a single, public function `decompress and write_to_gcs` would be useful for folks and would make the above code simple b/c you could just call that public api and pass in the compression type

Reply:
That's essentially the `unzip_blob()` function, no? You pass in `zip` or `gzip` and the private function is accessed under the hood

Reply:
yup you're totally right. this is good to go

Reply:
🎸 🎸 🎸